src/iceberg/json_internal.cc

```diff
@@ -519,8 +519,7 @@ Result<std::unique_ptr<PartitionField>> PartitionFieldFromJson(
                                           std::move(transform));
 }
 
-Result<std::unique_ptr<PartitionSpec>> PartitionSpecFromJson(
-    const std::shared_ptr<Schema>& schema, const nlohmann::json& json) {
+Result<std::unique_ptr<PartitionSpec>> PartitionSpecFromJson(const nlohmann::json& json) {
```

Comment:
Let's keep the function signature to still accept `const std::shared_ptr<Schema>& schema` and add a TODO comment below for PartitionSpec to validate the partition fields with this schema. Otherwise you need to add this parameter back in the next PR.